src/main/java/jenkins/plugins/gerrit/GerritApiBuilder.java

```diff
@@ -117,26 +117,23 @@ public GerritApiBuilder stepContext(StepContext context)
 
   public GerritApi build() {
     GerritApi gerritApi = null;
-    List<HttpClientBuilderExtension> extensions = new ArrayList<>();
-    GerritAuthData authData = null;
     if (gerritApiUrl == null) {
       logger.println("Gerrit Review is disabled no API URL");
-    } else if (username == null) {
-      logger.println("Gerrit Review is disabled no credentials");
-      authData = new AnonymousAuth(gerritApiUrl.toString());
     } else {
-      authData = new GerritAuthData.Basic(gerritApiUrl.toString(), username, password);
+      List<HttpClientBuilderExtension> extensions = new ArrayList<>();
+      GerritAuthData authData = null;
+      if (username == null) {
+        authData = new AnonymousAuth(gerritApiUrl.toString());
+      } else {
+        authData = new GerritAuthData.Basic(gerritApiUrl.toString(), username, password);
+      }
       if (Boolean.TRUE.equals(insecureHttps)) {
```

Comment:
this could not have been in the else case of username == null, as it should be affective in any case